docs/src/main/asciidoc/using-vertx.adoc

```diff
@@ -0,0 +1,334 @@
+
+{project-name} web resources support asynchronous processing and streaming results over https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events[server-sent events].
+
+=== Asynchronous processing
+
+Most programming guides start easy with a greeting service and this one makes no exception.
+
+To asynchronously greet a client, the endpoint method must return a `java.util.concurrent.CompletionStage`:
+
+[source,java]
+----
+@Path("/hello")
+public class GreetingResource {
+
+    @Inject Vertx vertx;
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    @Path("{name}")
+    public CompletionStage<String> greeting(@PathParam("name") String name) {
+        // When complete, return the content to the client
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        long start = System.nanoTime();
+
+        // TODO: asynchronous greeting
+
+        return future;
+    }
+}
+----
+
+So far so good.
+Now let's use the Vert.x API to implement the artificially delayed reply with the `setTimer` provided by Vert.x:
+
+[source,java]
+----
+// Delay reply by 10ms
+vertx.setTimer(10, l -> {
+    // Compute elapsed time in milliseconds
+    long duration = MILLISECONDS.convert(System.nanoTime() - start, NANOSECONDS);
+
+    // Format message
+    String message = String.format("Hello %s! (%d ms)%n", name, duration);
+
+    // Complete
+    future.complete(message);
+});
+----
+
+That's it.
+Now start {project-name} in `dev` mode with:
+
+[source, shell]
+----
+mvn compile quarkus:dev
+----
+
+Eventually, open your browser and navigate to http://localhost:8080/hello/Quarkus, you should see:
+
+[source, shell]
+----
+Hello Quarkus! (10 ms)
+----
+
+=== Streaming using Server-Sent Events
+
+{project-name} web resources that need to send content as https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events[server-sent events] must have a method:
+
+* declaring the `text/event-stream` response content type
+* returning a https://www.reactive-streams.org/[Reactive Streams] `Publisher`
+
+In practice, a streaming greeting service would look like:
+
+[source,java]
+----
+@Path("/hello")
+public class StreamingResource {
+
+    @GET
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    @Path("{name}/streaming")
+    public Publisher<String> greeting(@PathParam("name") String name) {
+        // TODO: create a Reactive Streams publisher
+        return publisher;
+    }
+}
+----
+
+How to create a Reactive Streams publisher?
+There are a few ways to do this:
+
+1. If you use `io.vertx.axle.core.Vertx`, the API provides `toPublisher` methods (and then use RX Java 2 or Reactive Streams Operators to manipulate the stream)
+2. You can also use `io.vertx.reactivex.core.Vertx` which already provides RX Java 2 (RX Java 2 `Flowable` implement Reactive Streams `publisher`).
+
+The first approach can be implemented as follows:
+
+[source, java]
+----
+// Use io.vertx.axle.core.Vertx;
+@Inject Vertx vertx;
```

Comment:
For this example it would be better to do the tick stream transformations with Reactive MP operators, I think. After all, the Axle shim is designed to work well with Quarkus and MP reactive APIs.

Reply:
Very good point, it's back!